src/Common/Config/ConfigProcessor.h

```diff
@@ -161,7 +163,9 @@ class ConfigProcessor
             Poco::XML::Node * node,
             zkutil::ZooKeeperNodeCache * zk_node_cache,
             const zkutil::EventPtr & zk_changed_event,
-            std::unordered_set<std::string> & contributing_zk_paths);
+            std::unordered_set<std::string> & contributing_zk_paths,
```

Comment:
Let's document what `current_path` is. It also seems like an important parameter (not sure), perhaps it should be moved further to the front.